### PR TITLE
test: add AIInsights empty fallback coverage

### LINKED ISSUE #2493: test(AIInsights-empty-fallback): verify Edge Cases & Empty/Missing Inputs Verification (Variation 1)
## 🧩 Background
The component/utility at `components/dashboard/AIInsights.tsx` needs isolated unit and integration testing. This issue focuses specifically on **Edge Cases & Empty/Missing Inputs Verification**. Handling missing options, unconfigured objects, or empty arrays is critical to guarantee component stability in production.

## 🎯 Objective
Create the new test file `components/dashboard/AIInsights.empty-fallback.test.tsx` targeting Edge Cases & Empty/Missing Inputs Verification and achieve complete coverage on the specified conditions.

## 📁 Files to touch
- `components/dashboard/AIInsights.empty-fallback.test.tsx` (new)

## 🛠️ Implementation steps
- Render the target module or component with empty arrays or null parameters.
- Verify that a clear, non-breaking fallback UI or error message is displayed.
- Verify standard styles are maintained in this default empty layout state.
- Assert that no unexpected runtime errors or hydration failures occur.
- Check key DOM structures to make sure empty markers exist.

## ✅ Definition of done
- [ ] 5 test cases.
- [ ] `vitest run` passes.

### PR DESCRIPTION
**Pull request by Srishti-Gupta74**
## Description

Fixes #2493

Added a new test file `AIInsights.empty-fallback.test.tsx` to verify edge cases and empty/missing input handling for the AIInsights component.

### Changes made

* Added test coverage for empty insights arrays.
* Added test coverage for `undefined` insights input.
* Added test coverage for `null` insights input.
* Verified that the component continues to render correctly under missing-input conditions.
* Fixed a runtime crash caused by calling `.map()` on `undefined` or `null` by safely handling missing insights data with a fallback array.

## Pillar

* [ ] 🎨 Pillar 1 — New Theme Design
* [ ] 📐 Pillar 2 — Geometric SVG Improvement
* [ ] 🕐 Pillar 3 — Timezone Logic Optimization
* [x] 🛠️ Other (Bug fix, refactoring, docs)

## Visual Preview

N/A (test coverage and stability improvement)

## Checklist before requesting a review:

* [x] I have read the `CONTRIBUTING.md` file.
* [ ] I have tested these changes locally (`localhost:3000/api/streak?user=YOUR_USERNAME`).
* [ ] I have run `npm run format` and `npm run lint` locally and resolved all errors (CI will fail otherwise).
* [x] My commits follow the Conventional Commits format (e.g., `feat(themes): ...`, `fix(calculate): ...`).
* [ ] I have updated `README.md` if I added a new theme or URL parameter.
* [x] I have started the repo.
* [x] I have made sure that i have only one commit to merge in this PR.
* [x] The SVG output matches the CommitPulse "premium quality" aesthetic standard (no raw elements, smooth animations, correct fonts).
* [x] (Recommended) I joined the CommitPulse Discord community for contributor discussions, mentorship, and faster PR support.

### CODE REVIEW

File: components/dashboard/AIInsights.empty-fallback.test.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+// Mock lucide-react
+vi.mock('lucide-react', () => ({
+  Sparkles: (props: any) => <div data-testid="icon-sparkles" {...props} />,
+  Moon: (props: any) => <div data-testid="icon-moon" {...props} />,
+  Sun: (props: any) => <div data-testid="icon-sun" {...props} />,
+  Zap: (props: any) => <div data-testid="icon-zap" {...props} />,
+  Calendar: (props: any) => <div data-testid="icon-calendar" {...props} />,
+  Flame: (props: any) => <div data-testid="icon-flame" {...props} />,
+  Code: (props: any) => <div data-testid="icon-code" {...props} />,
+  Star: (props: any) => <div data-testid="icon-star" {...props} />,
```
[the remark this answers is not in the document]
Comment (Srishti-Gupta74) [reply components/dashboard/AIInsights.empty-fallback.test.tsx:39]: The tests intentionally verify runtime resilience against invalid/missing inputs. The component API remains unchanged.

File: components/dashboard/AIInsights.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ export default function AIInsights({ insights }: { insights: AIInsight[] }) {
       </div>
 
       <div className="flex flex-col gap-6">
-        {insights.map((insight, i) => {
+        {(insights ?? []).map((insight, i) => {
```
[the remark this answers is not in the document]
Comment (Srishti-Gupta74) [reply components/dashboard/AIInsights.tsx:26]: This PR focuses on runtime safety and edge-case coverage. Adjusting the public prop contract would broaden the scope beyond the issue requirements.

File: components/dashboard/AIInsights.empty-fallback.test.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  it('renders only the header sparkles icon when insights array is empty', () => {
+    render(<AIInsights insights={[]} />);
+
+    const sparklesIcons = screen.getAllByTestId('icon-sparkles');
+    expect(sparklesIcons).toHaveLength(1);
+  });
+
+  it('renders heading when insights is undefined', () => {
+    render(<AIInsights insights={undefined as unknown as AIInsight[]} />);
```
[the remark this answers is not in the document]
Comment (Srishti-Gupta74) [reply components/dashboard/AIInsights.empty-fallback.test.tsx:62]: The cast is intentional to verify runtime resilience against invalid/missing inputs as required by the issue.

File: components/dashboard/AIInsights.empty-fallback.test.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    vi.clearAllMocks();
+  });
+
+  it('renders with an empty insights array', () => {
+    render(<AIInsights insights={[]} />);
+
+    expect(screen.getByText('AI Insights')).toBeDefined();
+  });
+
+  it('renders only the header sparkles icon when insights array is empty', () => {
```
[the remark this answers is not in the document]
Comment (Srishti-Gupta74) [reply components/dashboard/AIInsights.empty-fallback.test.tsx:54]: Kept individual test cases for clarity and to keep each edge case independently documented.

**Review by Aamod007 — APPROVED** [review]
